src/test/java/org/broadinstitute/hellbender/tools/dragstr/CalibrateDragstrModelIntegrationTest.java

```diff
@@ -11,26 +17,72 @@
+            new File("src/test/resources/org/broadinstitute/hellbender/tools/dragstr/human-b37-decimated-20-21.strtbl");
+    private static final File EXPECTED_PARAMS_FILE =
+            new File("src/test/resources/org/broadinstitute/hellbender/tools/dragstr/expected-ceutrio-hiseq-wgs-b37-20-21.drgstr");
 
+    @DataProvider
+    public static Object[][] threadCountAndBamCramData() {
+        return new Object[][]{ {1, false}, {1, true}, {3, true}, {3, false}, {4, false}, {4, true}};
+    }
 
-    // Asserting that ReadsDataSourcePool passes the reference through to the underlying reader
-    @Test()
-    public void testReferenceGetsProvidedToCramInputWhenThreaded() throws IOException {
-        final File output = File.createTempFile("test-output", ".zip");
+    //TODO Ideally the bam/cram would be large enough so that forcing the estimation is not necessary.
+    //TODO we should cosider not only human but also other species sample that are nore compect and
+    //TODO with larger propr
+    @Test(dataProvider = "threadCountAndBamCramData")
+    public void testDragstrModelInferenceForcingEstimation(final int threads, final boolean useCram) throws IOException {
+        final File output = File.createTempFile("test-output", ".dragstr    ");
         output.deleteOnExit();
 
         final ArgumentsBuilder args = new ArgumentsBuilder();
         args.add(StandardArgumentDefinitions.REFERENCE_SHORT_NAME, b37_reference_20_21);
         args.add(CalibrateDragstrModel.STR_TABLE_PATH_FULL_NAME, HUMAN_B37_20_21_TBL_GZ);
         args.addFlag(CalibrateDragstrModel.PARALLEL_FULL_NAME);
-        args.add(CalibrateDragstrModel.THREADS_FULL_NAME, 10);
+        if (threads > 1) args.add(CalibrateDragstrModel.THREADS_FULL_NAME, threads);
+        if (threads == 0) args.addFlag(CalibrateDragstrModel.PARALLEL_FULL_NAME);
         args.add(CalibrateDragstrModel.SHARD_SIZE_FULL_NAME,  CalibrateDragstrModel.DEFAULT_SHARD_SIZE);
         args.add(CalibrateDragstrModel.DOWN_SAMPLE_SIZE_FULL_NAME, CalibrateDragstrModel.DEFAULT_DOWN_SAMPLE_SIZE);
-        args.addInput(NA12878_20_21_WGS_cram);
+        args.addInput(useCram ? NA12878_20_21_WGS_cram : NA12878_20_21_WGS_bam);
+        args.addFlag(CalibrateDragstrModel.FORCE_ESTIMATION_FULL_NAME);
         args.addOutput(output);
+        runCommandLine(args);
+
+        final DragstrParams actual = DragstrParamUtils.parse(new GATKPath(output.toString()));
+        final DragstrParams expected = DragstrParamUtils.parse(new GATKPath(EXPECTED_PARAMS_FILE.toString()));
+        assertEquals(actual, expected);
+    }
 
+    @Test(dataProvider = "threadCountAndBamCramData")
+    public void testDragstrModelInferenceFailingOverToDefaults(final int threads, final boolean useCram) throws IOException {
+        final File output = File.createTempFile("test-output", ".drgstr");
+        output.deleteOnExit();
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
```

Comment:
Per their documentation and personal mail exchange with Arun is an all-or-nothing, if any combo fails we simply use the default tables for all periods... 
```
3.5 Default calibration values

When there is insufficient STR loci coverage (e.g. in targeted sequencing like exomes and panels), the calibration algorithm is skipped and default values for GOP, GCP and API are used.

Calibration is enabled if there are at least 200 qualifying locations in each of the following STR bins:
```
Arun's answer (quoted):
```
Yes, it is all or nothing. All (period,length) combos have to have at least the minimum number of cases. If any of them has less than minimum number, default tables are used.
```
However what you are saying makes perfect sense since the final period estimates seems to be independent.  

I guess that they might be worried that if one uses defaults for particular periods and empirical calibrated values for other then you end up with "Frankstein" tables that  may not make sense... for example period 5 str artifacts are more likely than period 4 if period 5 is default and the sample happens to be PCR-free. 

A better solution would be to generate defaults based on the estimated ones by extrapolation/extension or interpolation.

I would keep it simple for now and as as they say.